runtime/server/server.go

```diff
@@ -272,6 +270,9 @@ func timeoutSelector(fullMethodName string) time.Duration {
 func errorMappingUnaryServerInterceptor() grpc.UnaryServerInterceptor {
 	return func(ctx context.Context, req interface{}, info *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (interface{}, error) {
 		resp, err := handler(ctx, req)
+		if _, ok := status.FromError(err); ok {
+			return resp, err
+		}
 		return resp, mapGRPCError(err)
```

Comment:
nit: implement this check inside `mapGRPCError` – see how it's done in `admin/server/server.go`